swagger.yaml

```diff
@@ -0,0 +1,221 @@
+          schema:
+            $ref: '#/definitions/JobState'
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}/filter:
+    put:
+      tags:
+      - "Jobs"
+      summary: "Add a dimension to filter the dataset"
+      description: "The dimension can only be added into the job if the state is still set to `created` otherwise 400 status code is returned"
+      produces:
+      - "application/json"
+      parameters:
+      - $ref: '#/parameters/jobId'
+      - $ref: '#/parameters/dimension'
+      responses:
+        200:
+          description: "Filter list was updated"
+        400:
+          description: "The job has been locked as it has been `submitted` to be processed"
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+  /jobs/{jobId}/state:
+    put:
+      tags:
+      - "Jobs"
+      summary: "Update the state of the job"
+      description: |
+        The job state can only be updated if the state is still `created`. If the state is changed to `submitted`,
+        this will lock the job.
+      parameters:
+      - $ref: '#/parameters/jobId'
+      - $ref: '#/parameters/state'
+      responses:
+        200:
+          description: "The state has been updated"
+        400:
+          description: "The job has been locked as it has been `submitted` to be processed"
+        404:
+          description: "Job not found"
+        500:
+          description: "Failed to process the request due to an internal error"
+definitions:
+  JobInstance:
+    type: object
+    properties:
+      id:
+        type: string
+        description: "An unique id for the filter job"
+  NewJob:
+    description: |
+      A model used to create a new filter job
+    type: object
+    required: ["dataset", "edition", "version"]
+    properties:
+      dataset:
+        type: string
+        description: "The type of dataset"
+      edition:
+        type: string
+        description: "The edtion of the dataset (This could be monthly or yearly version)"
+      version:
+        type: string
+        description: "The version of the dataset (This includes any amendments to the original dataset edition)"
+      filters:
+        type: array
+        items:
+          $ref: '#/definitions/Dimension'
+  JobState:
+    description: |
+      A description of a job to generate a customised dataset
+    type: object
+    properties:
+      dataset:
+        type: string
+        description: "The type of dataset"
+      edition:
+        type: string
+        description: "The edtion of the dataset (This could be monthly or yearly version)"
+      version:
+        type: string
+        description: "The version of the dataset (This includes any amendments to the original dataset edition)"
```

Comment:
I am a bit confused about how dimensions are defined. I think what you call dimensions I think of as dimension options. For example, a dimension would be gender, with the options male and female. What we are adding here are dimension options.